modules/internal/ChapelRange.chpl

```diff
@@ -215,12 +215,21 @@ module ChapelRange {
     compilerError("Bounds of '..' must be integers of compatible types, when specified.");
   }
 
-  // Range builders for partially bounded ranges
-  proc chpl_build_partially_bounded_range(param bt: BoundedRangeType, bound: int(?w))
+  // Range builders for low bounded ranges
+  proc chpl_build_low_bounded_range(param bt: BoundedRangeType, bound: int(?w))
     return new range(int(w), bt, false, bound, bound);
-  proc chpl_build_partially_bounded_range(param bt: BoundedRangeType, bound: uint(?w))
+  proc chpl_build_low_bounded_range(param bt: BoundedRangeType, bound: uint(?w))
     return new range(uint(w), bt, false, bound, bound);
-  proc chpl_build_partially_bounded_range(param bt: BoundedRangeType, bound) {
+  proc chpl_build_low_bounded_range(param bt: BoundedRangeType, bound) {
+    compilerError("Bounds of '..' must be integers of compatible types, when specified.");
```

Comment:
Should this phrasing simply say "must be integers" since there won't be two of them in low/high-bounded ranges (so it's not clear what "compatible" types would refer to")?  I guess arguably that's what the "when specified" covers, but since we know that one was not specified in these cases (and that both were specified in the build_bounded_range() case), would the error messages be clearer if they were tuned to the specific case?  (i.e., the bounded case could drop the ", when specified" since they are and the unbounded case could simply say "must be integers.")

I guess one argument might be that the unified error messages across all cases teaches something about the language, but I think it arguably does so at the expense of less clear messages.

I'm not insistent on this, but I think it's worth consideration.